server/aesdsocket.c

```diff
@@ -0,0 +1,341 @@
+		printf("Error: Binding with address failed\n");
+		syslog(LOG_ERR, "Error while binding socket= %s. Exiting.", strerror(errno));
+		freeaddrinfo(res);
+		exit(EXIT_FAILURE);
+	}
+
+	// All recived data should be stored in the PATH file_data
+	int file_fd = creat(file_data, 0644); // creating file if it doesn't exist with 644 permissions
+	if (file_fd == -1)					  // generating error
+	{
+		printf("Error while creating file \n");
+		syslog(LOG_ERR, "Error: File could not be created!= %s. Exiting...", strerror(errno));
+		exit(EXIT_FAILURE);
+	}
+	close(file_fd);
+
+	// free res when no longer needed
+	freeaddrinfo(res);
+
+	// Step-4 Listening and Accepting packets in a loop
+	// Loop for listening and accepting data from client
+	while (1)
+	{
+
+		output_buffer = (char *)malloc(BUFFER_SIZE); // mallocing output buffer to send data to file
+		if (output_buffer == NULL)
+		{
+			printf("buffer not created\n");
+			exit(EXIT_FAILURE);
+		}
+		memset(output_buffer, 0, BUFFER_SIZE);
+
+		// Listening for client
+		int temp_listen = listen(socket_fd, MAX_BACKLOG);
+		if (temp_listen == -1) // generating error
+		{
+			printf("Error while listening \n");
+			syslog(LOG_ERR, "Error: Listening failed =%s. Exiting ", strerror(errno));
+			exit(EXIT_FAILURE);
+		}
+
+		client_size = sizeof(struct sockaddr);
+
+		// Accepting connection
+		accept_fd = accept(socket_fd, (struct sockaddr *)&client_add, &client_size);
+		if (accept_fd == -1) // generating error
+		{
+			printf("Error while accepting \n");
+			syslog(LOG_ERR, "Error: Accepting failed =%s. Exiting ", strerror(errno));
+			exit(EXIT_FAILURE);
+		}
+		// to get the client address in a readable format
+		struct sockaddr_in *addr_in = (struct sockaddr_in *)&client_add;
+		char *addr_ip = inet_ntoa(addr_in->sin_addr); // using inet_ntoa function
+
+		syslog(LOG_DEBUG, "Connection succesful. Accepting connection from %s", addr_ip);
+		printf("Connection succesful.Accepting connection from %s\n", addr_ip);
+
+		int i;
+		bool receive = false; // set the completion of packets to stop completion when
+
+		while (receive == false)
+		{
+
+			int temp_recv = recv(accept_fd, buff, BUFFER_SIZE, 0);
+			if (temp_recv < 0)
+			{
+				printf("Error while receving data packets\n");
+				syslog(LOG_ERR, "Error: Receiving failed =%s. Exiting ", strerror(errno));
+				exit(EXIT_FAILURE);
+			}
+			else if (temp_recv == 0)
+			{ // if no bytes received it will return 0
+				break;
+			}
+
+			for (i = 0; i < BUFFER_SIZE; i++)
+			{
+				if (buff[i] == '\n') // when \n received means end of file
+				{
+					receive = true; // make the bool true to end the receiving loop
+					i++;
+					printf("data packet receiving completed\n");
+					syslog(LOG_DEBUG, "data packet received");
+					break;
+				}
+			}
+			data_count += i; // increase data count for increasing packet size
+
+			// reallocate the buffer so as to accomodate the new data count
+			output_buffer = (char *)realloc(output_buffer, (data_count + 1));
```

Comment:
@ayswariya088 I think this will actually allocate the buffer to the full size of all packets sent since startup I believe, not just the number of packets sent this connection.